src/ntrip_client/nmea_parser.py

```diff
@@ -15,11 +15,11 @@ def __init__(self, logerr=logging.error, logwarn=logging.warning, loginfo=loggin
 
   def is_valid_sentence(self, sentence):
     # Simple sanity checks
-    if len(sentence) > _NMEA_MAX_LENGTH:
+    if len(sentence.rstrip()) > _NMEA_MAX_LENGTH:
```

Comment:
All documentation I can find on the NMEA spec says that the max length for a sentence is 82 characters including the `$` and `\r\n`. If you need the parser to support longer sentences, perhaps this should be a parameter to the `NMEAParser`.

See:
https://gpsd.gitlab.io/gpsd/NMEA.html
https://en.wikipedia.org/wiki/NMEA_0183

Reply:
The standard is 82 characters as you say. The ZED-F9P, in high precision mode can actually have more (i am struggling to find out how many more, but never seen more than 82 EXCLUDING \r\n...) see page 7 first table [](https://cdn.sparkfun.com/assets/f/7/4/3/5/PM-15136.pdf). Also other vendors have something similar. 
It would be a shame to have the high precision mode truncated if we want to use the same string for other porpoises (more than enough for the ntrip server though...)

Reply:
Yes, I agree on removing `.rstrip()` . I already did. (see latest commits). I like your idea, but was wondering if instead of trowing a debug message (somewhat hidden), what if we just leave the limit at 82 and log an error message explaining what might be going on and how it can be fixed? In such case we could also use a parameter and leaving it "hidden" unless required.. Just an idea, whatever you like the most.

Reply:
Yeah I like that solution, we could default the max length to 82 if `nmea_max_length` is not set, otherwise use whatever that is set to, and then in the launch file default `nmea_max_length` to 82 as well and document why someone might change the value

Reply:
Done